script/build.js

```diff
@@ -119,36 +110,48 @@ smith.use(layouts({
   pattern: '**/*.{md,html}'
 }));
 
+/*
+Add nonce attribute with substition string to all inline script tags
+Convert onclick event handles into nonced script tags
+*/
+smith.use(nonceTransformer);
+
+// Create the data passed from the content build to the assets compiler.
+// On the server, it can be accessed at BUILD_OPTIONS.buildSettings.
+// In the browser, it can be accessed at window.settings.
+smith.use(createBuildSettings(BUILD_OPTIONS));
+
+if (BUILD_OPTIONS.watch) {
+  const watchPaths = { [`${BUILD_OPTIONS.contentRoot}/**/*`]: '**/*.{md,html}' };
+  const watchMetalSmith = watch({ paths: watchPaths, livereload: true });
+  smith.use(watchMetalSmith);
+  smith.use(webpackMetalsmithConnect.watchAssets(BUILD_OPTIONS));
+} else {
+  smith.use(webpackMetalsmithConnect.compileAssets(BUILD_OPTIONS));
+
+  const isDevBuild = [environments.DEVELOPMENT, environments.VAGOVDEV].includes(BUILD_OPTIONS.buildtype);
+  if (!isDevBuild) {
+    smith.use(addAssetHashes(BUILD_OPTIONS));
+  }
+
+  if (process.env.CHECK_BROKEN_LINKS !== 'no') {
+    smith.use(checkBrokenLinks(BUILD_OPTIONS));
+  }
+}
+
```

Comment:
I shifted around the middleware registrations around so that `createBuildSettings` was called before `watch`. This is so build-settings can be passed to Webpack Dev Server.